shairport.c

```diff
@@ -2102,6 +2108,18 @@ int main(int argc, char **argv) {
   // parse arguments into config -- needed to locate pid_dir
   int audio_arg = parse_options(argc, argv);
 
+#ifdef COMPILE_FOR_OPENBSD
+  /* Drop "proc exec" unless external commands are to be run. */
+  if ((config.cmd_active_start == NULL) &&
+      (config.cmd_active_stop == NULL) &&
+      (config.cmd_start == NULL) &&
+      (config.cmd_stop == NULL) &&
+      (config.cmd_set_volume == NULL)) {
+    if (pledge("stdio rpath wpath cpath dpath inet unix dns audio", NULL) == -1)
+      die("pledge");
+  }
+#endif
```

Comment:
To limit the dangerous `"proc exec"` set to only those commands, one had to hoist `poptParseArgvString()` out of common.c`s command handlers, so the following becomes possible:
```
  } else {
    unveil(parsed_active_start_argv0, "x");
    ....
    unveil("/", "rwc");
    unveil(NULL, NULL);  // or pledge("..." /* no "unveil" */, NULL);
  }
```